tsl/src/compression/compression.c

```diff
@@ -2531,18 +2532,21 @@ find_matching_index(Relation comp_chunk_rel, List **index_filters, List **heap_f
 
 /*
  * This method will evaluate the predicates, extract
- * left and right operands, check if one of the operands is
- * a simple Var type. If its Var type extract its corresponding
- * column name from hypertable_compression catalog table.
- * If extracted column is a SEGMENT BY column then save column
- * name, value specified in the predicate. This information will
- * be used to build scan keys later.
+ * left and right operands, check if any of the operands
+ * can be used for batch filtering and if so, it will
+ * create a BatchFilter object and add it to the corresponding
+ * list.
+ * Any segmentby filter is put into index_filters list other
+ * filters are put into heap_filters list.
  */
 static void
 process_predicates(Chunk *ch, CompressionSettings *settings, List *predicates, List **heap_filters,
 				   List **index_filters, List **is_null)
 {
 	ListCell *lc;
+	PlannerGlobal glob = { .boundParams = NULL };
```

Comment:
Should we take `es_param_list_info` from the node state here, like in runtime chunk exclusion? Then it could work for prepared statement params, and under joins.

Reply:
I think for joins it would not be safe as we dont repeat decompression in rescan. For now this is intentional. I'll add some more tests cause having prepared statement support would be nice.

Reply:
Oh, right, I think it's not even possible, because the rescans happen underneath the DELETE, and it receives the tids only. And we have to guess which batches we have to decompress even before the underlying nested loop starts running.

Would be nice if the underlying scans could return the matching compressed tids as well, and then we'd use that to decompress, but that's a completely different approach.

Reply:
Might also create additional weirdness with the snapshots betweeen the internal scan and the normal scan

Reply:
Prepared Statement works too as they are handled differently from join params, i'll add a test

Reply:
I've added a comment as well.